### PR TITLE
Sharing

### PR DESCRIPTION
**Pull request by charpath**
Implement sharing of tasks into the app. Allow users to share a given task with any other user using their username.

### CODE REVIEW

File: packages/react-frontend/src/components/TableTask.jsx [hunk 1]
```diff
@@ -29,6 +40,19 @@ function TableTask(props) {
                 <Tag className="date-tag" icon="pi pi-calendar">
                     {new Date(task.due_date_time).toLocaleDateString("en-US")}
                 </Tag>
+                <button className="delete-btn" onClick={() => setEditDialogVisible(true)}>
```
Comment (Ryan-Ficklin) [inline packages/react-frontend/src/components/TableTask.jsx:43]: This code is duplicates our edit and remove buttons. The div just below handles edit and remove, these buttons are redundant

**Review by Ryan-Ficklin — CHANGES_REQUESTED** [review]

File: packages/react-frontend/src/components/TableTask.jsx (hunk 1 above)
Comment (charpath) [reply packages/react-frontend/src/components/TableTask.jsx:43]: Are you proposing removing code between lines 43 and 55? I don't see any duplicate buttons, those lines of code should create one copy of the edit button, one copy of the share button, and one copy of the delete button

Comment (charpath) [reply packages/react-frontend/src/components/TableTask.jsx:43]: Never mind, I see the problem now. The addition of the icons for edit and delete was not present on my dev branch. Will fix and update

**Comment by jfwoo** [comment]
no comments, looks good to me

**Review by Ryan-Ficklin — COMMENTED** [review]
The update looks good. I like the new look of the share button.

**Review by Ryan-Ficklin — APPROVED** [review]

**Comment by pkrishna05** [comment]
Looks good

**Comment by siddarthmaddhula** [comment]
Everything looks good here.

**Pull request merged**